tutorial/index.html

```diff
@@ -0,0 +1,412 @@
+        In the previous section we described how to define an actor as a Reactive Streams Publisher. To start streaming
+        we still need to instantiate the actor and - in one go - transform it into a Source. This is done in <a
+            href="#code/src/main/scala/reactive/tweets/outgoing/TweetFlow.scala" class="shortcut">TweetFlow.scala</a>.
+
+    <pre><code>private val tweetSource: Source[Tweet, ActorRef] =
+        Source.actorPublisher[Tweet](TweetPublisher.props)</code></pre>
+
+    Now we've got our hands on a proper <code>Source</code> we can use it to construct a <code>Flow</code>:
+
+        <pre><code>def tweetFlow(tweetFilter: TweetFilter): Flow[Message, Message, Unit] =
+            Flow.wrap(Sink.ignore, tweetSource filter tweetFilter map toMessage)(Keep.left)</code></pre>
+
+    Note: ignore the tweetFilter for now, this is just a convenience parameter to be able to construct specific streams,
+    like the stream of tweets for a specific user, or the stream of tweets that contain a specific hash tag.
+    </p>
+
+    <p>
+        We are only interested in one way communication with our Websocket. We push new tweets to the client on their
+        arrival. Theoretically we could also receive messages over the same channel from the client. We just ignore
+        those.
+        So what does this look like in terms of a Flow?
+    </p>
+
+    <pre>
+        +-----------------------------+
+        | Our Websocket Flow          |
+        |                             |
+        |  +--------+      +------+   |
+        |   \       |      |       \  |
+    In ~~>   | SINK |      | SOURCE |~~> Out
+        |   /       |      |       /  |
+        |  +--------+      +------+   |
+        +-----------------------------+
+    </pre>
+
+    <p>
+        The important thing to note is that the inner parts of the Flow (the Sink and the Source) have no connection
+        with each other. Incoming messages from the Websocket are ignored by routing them to the Ignore Sink
+        while on the other hand the outgoing channel will be serviced by our transformed actor that is listening to the
+        event bus:
+
+    <pre><code>Flow.wrap(Sink.ignore, tweetSource filter tweetFilter map toMessage)(Keep.left)</code></pre>
+    </p>
+    <p>
+        Next: <a href="#tutorial/6" class="shortcut">The Websocket</a>
+    </p>
+</div>
+
+<div>
+    <h2>Part 1: The Websocket</h2>
+
+    <p>
+        So we have a nice Flow, the next step is plugging it into Akka Http.<br/>
+        The most notable part is the <code>handleWebsocketMessages</code> directive, which comes out of the box with Akka Http.
+<pre><code>def tweetsOfUserSocket = (pathPrefix("users") & path(Segment)) { userName =>
+  handleWebsocketMessages(tweetFlowOfUser(userName))
+}</code></pre>
+
+    The part below describes the path at which the Websocket for this Flow can be found: <code>/ws/users/[userName]</code>.
+
+<pre><code>pathPrefix("ws") {
+  pathPrefix("tweets") {
```

Comment:
The path should be `/ws/users/[userName]`. Where does this `tweets` pathprefix show up?